org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLFormatter.java

```diff
@@ -344,6 +327,26 @@ private static boolean isPreviousSiblingNodeType(DOMNode node, short nodeType) {
 		return previousNode != null && previousNode.getNodeType() == nodeType;
 	}
 
+	private static void addPIToXMLBuilder(DOMNode node, XMLBuilder xml) {
+		DOMProcessingInstruction processingInstruction = (DOMProcessingInstruction) node;
+		xml.startPrologOrPI(processingInstruction.getTarget());
+		if (processingInstruction.hasAttributes()) {

```

Comment:
A processing instruction will never have attributes, it's always text content (the case below).

Reply:
What case are you referring to?